Improve right-click behaviour in table view

jclarkeSTFC opened the pull request:
If you right-click on a row or column header in the table view (e.g. "Show Data"), then that row or column will be selected.

**Description of work**
In the methods that set up the header context menus, the relevant row/column is now selected.

**Purpose of work**
<!-- Why has this work been done? If there is no linked issue please provide appropriate context for this work.
-->
Fixes #35583 

<!-- If the original issue was raised by a user they should be named here. Do not leak email addresses
**Report to:** [user name]
-->

**Summary of work**
<!-- Please provide a short, high level description of the work that was done.
-->
When right-clicking on a row or column header in the table view, that whole row/column will be selected before opening the context menu. This stops you right-clicking on a row/column header then accidentally exporting some data selected in another place, i.e. the assumption now is that if you right-click on a header, then you want the data in that row/column.

**To test:**

<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->
- See #35583 for original issue
- Maybe try selecting some data, then right clicking in various places. Export, copy, plot, etc from the context menu should all still work, (and do their thing on the correct numbers).

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

jclarkeSTFC committed:
Select current row/column on right-click

If you right-click on a row or column header in the
table view (e.g. "Show Data"), then that row or column
will be selected.

jclarkeSTFC committed:
Added release notes

jhaigh0 reviewed (changes requested):
If you select multiple columns / rows and right-click on the selected headers, it changes to select only the one right-clicked.

jclarkeSTFC committed:
Handle multi-selected rows/columns.

When multiple rows/columns are selected, then if the user
right-clicks inside one of those headers there's no need
to select anything else.

jhaigh0 reviewed (approved):
Good small bug fix that makes the matrix view easier to use.
